### PR TITLE
Nucleo f429 zi display driver

### PR DESCRIPTION
**Pull request by Cristian243342**
### Pull Request Overview

Add a new graphic display capsule for the board Nucleo-F429ZI (and it's dependencies), which fixes a bug that caused the first column of pixels on the board display to be dead. The implementation is unfinished due to a bug with the boards/components/src/bus.rs i2c_master_bus_component_static! macro (opened separate pull request [tock#3810](https://github.com/tock/tock/pull/3810); waiting for review).

### Testing Strategy

This pull request is untested since it is unfinished.

### TODO or Help Wanted

This pull request still needs:
- [ ] Modifying/utilizing the existing screen.rs capsule with it's components instead of using a separate graphics_display.rs capsule.
- [ ] Initialization of the graphic display capsule's i2c bus.
- [ ] Update documentation.
- [ ] Uncomment commented code and test implementation.

### Documentation Updated

-[  ] None.

### Formatting

- [ ] Ran `make prepush`.

### CODE REVIEW

File: capsules/extra/src/bus.rs [hunk 1]
```diff
@@ -301,7 +301,21 @@ impl<'a, I: I2CDevice> Bus<'a> for I2CMasterBus<'a, I> {
...
+                    buffer[1] = addr as u8;
+                    self.status.set(BusStatus::SetAddress);
+                    match self.i2c.write(buffer, 2) {
+                        Ok(()) => Ok(()),
+                        Err((error, buffer)) => {
+                            self.addr_buffer.replace(buffer);
+                            Err(error.into())
+                        }
+                    }
+                }),
```
Comment (alexandruradovici) [inline capsules/extra/src/bus.rs:318]: Please split this in a new pull reqeuest, as it modifies the the `bus` library.

**Review by alexandruradovici — CHANGES_REQUESTED** [review]

File: capsules/extra/src/bus.rs (hunk 1 above)
Comment (Cristian243342) [reply capsules/extra/src/bus.rs]: Created a separate pull request [tock#3812](https://github.com/tock/tock/pull/3812) with the `bus` library modifications.

**Comment by lschuermann** [comment]
Marking as blocked on #3812.

**Comment by lschuermann** [comment]
Closing due to staleness of #3812, feel free to re-open when there's progress.

**Pull request closed without merging**